app/models/shift.rb

```diff
@@ -2,4 +2,26 @@ class Shift < ActiveRecord::Base
   belongs_to :giver, class_name: :User, foreign_key: :giver_id
   belongs_to :taker, class_name: :User, foreign_key: :taker_id
   has_many :offers
+
+  has_one :team, through: :giver, source: :team
+
+  def yes_offers?
+    covered = nil
+    self.offers.each do |offer|
+      covered = true if offer.type == "yes"
+    end
+      covered == true ? self.covered = true : self.covered = false
+  end
+
+  def self.team_shifts(current_user)
+    team_shifts = []
+    @shifts = Shift.all
```

Comment:
My main concern here is you load all shifts, which will become very slow as the app grows beyond a few hundred shift records. Also (and this is a minor point), you are returning an `Array`, when you could be returning an `ActiveRecord::Relation`. By returning an `ActiveRecord::Relation`, you give the client that called this method a more powerful structure.

You can solve both of these problems by using 2 queries as follows:

```ruby
def self.team_shifts(user)
  team_mate_ids = User.where(team_id: user.team_id).pluck(:id)
  Shift.where(giver_id: team_mate_ids)
end
```

Both queries in this method will scale much better than `Shift.all`. And by returning `Shift.where(...)`, the client that that called this gets all of the extra power of an `ActiveRecord::Relation`.

Going beyond your initial question, you could move these queries around to better adhere to the single responsibility principle. A shift needn't know anything about teams or how to find a giver's team mates. Your interface looks like

```ruby
shifts = Shift.team_shifts(current_user)
```

I suggest

```ruby
shifts = current_user.team.shifts
```

This usage suggests a user knows her team and a team knows its shifts. That feels better contained. This interfaces allows you to remove your method and use more built-in ActiveRecord magic

```ruby
class Team < ActiveRecord::Base
  has_many :users
  has_many :shifts, through: :users, source: :shifts_given  
end
```ruby